### PR TITLE
Jetty 9.4.x 2059 make osgi tests work jre9

### PR DESCRIPTION
**Pull request by janbartel**
Makes appropriate tests work in jdk9, plus adds a test for the alpn-java-server/client.

### CODE REVIEW

File: jetty-osgi/test-jetty-osgi/pom.xml [hunk 1]
```diff
@@ -532,9 +532,27 @@
...
+        <dependency>
+          <groupId>org.eclipse.jetty</groupId>
+          <artifactId>jetty-alpn-conscrypt-client</artifactId>
+          <version>${project.version}</version>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>org.conscrypt</groupId>
+          <artifactId>conscrypt-openjdk-uber</artifactId>
+          <version>1.0.0.RC11</version>
```
Comment (sbordet) [inline jetty-osgi/test-jetty-osgi/pom.xml]: Use here `${conscrypt.version}` that is defined in the main POM.

File: jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        res.add(mavenBundle().groupId("org.eclipse.jetty").artifactId("jetty-alpn-server").versionAsInProject().start());
+
+        res.add(mavenBundle().groupId("org.eclipse.jetty.http2").artifactId("http2-common").versionAsInProject().start());
+        res.add(mavenBundle().groupId("org.eclipse.jetty.http2").artifactId("http2-hpack").versionAsInProject().start());
+        res.add(mavenBundle().groupId("org.eclipse.jetty.http2").artifactId("http2-server").versionAsInProject().start());
+        return res;
+    }
+ 
+
+    @Ignore
```
Comment (sbordet) [inline jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java]: Do you need this `@Ignore` here ?

File: jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+        }
+        finally
+        {
+            if (httpClient != null) httpClient.stop();
+            if (http2Client != null) http2Client.stop();
+        }
+    }
+
+}
```
Comment (sbordet) [inline jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java]: If you can remove all the unneeded empty lines between methods, at the end of methods, etc. - looks tidier without them.

File: jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java (hunk 2 above)
Comment (janbartel) [reply jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2JDK9.java]: Yes, this test method is only ever used to help diagnose what has gone wrong if the actual test (testHTTP2) fails.

**Pull request merged**